python_modules/libraries/dagster-dbt/dagster_dbt/dbt_project.py

```diff
@@ -0,0 +1,212 @@
+import logging
+import os
+import shutil
+from pathlib import Path
+from typing import Optional, Sequence, Union
+
+from dagster._annotations import experimental, public
+from dagster._model import DagsterModel
+
+from .errors import DagsterDbtManifestNotFoundError, DagsterDbtProjectNotFoundError
+
+logger = logging.getLogger("dagster-dbt.artifacts")
+
+
+def using_dagster_dev() -> bool:
+    return bool(os.getenv("DAGSTER_IS_DEV_CLI"))
+
+
+def parse_on_load_opt_in() -> bool:
+    return bool(os.getenv("DAGSTER_DBT_PARSE_PROJECT_ON_LOAD"))
+
+
+@experimental
+class DbtManifestHandler:
+    """Life-cycle callbacks for dbt manifest represented by DbtProject."""
+
+    def on_load(self, project: "DbtProject"): ...
+    def on_prepare_for_deployment(
+        self,
+        project: "DbtProject",
+        *,
+        log_level: int = logging.INFO,
+    ): ...
+
+
+@experimental
+class DagsterDbtManifestHandler(DbtManifestHandler):
+    def __init__(
+        self,
+        generate_cli_args: Sequence[str] = ("parse", "--quiet"),
```

Comment:
I understand why we would want to do this if it were a `list` but since its an immutable `tuple` I thought this approach was preferable

Reply:
Hmm I'm just thinking about this case:

```python
In [2]: ("1", "2") == ["1", "2"]
Out[2]: False
```

Even though `generate_cli_args` isn't exposed, I feel like the majority of folks would pass them in as a list, so we should ensure our defaults match with that expectation. (Just in case if they do any equality checks with `DagsterDbtManifestHandler._generate_cli_args` or something)